Update README for 3.0.0 (#307)
microsoft/FeatureManagement-Dotnet: README.md

``` C#
using Microsoft.FeatureManagement;

public class Startup
{
  public void ConfigureServices(IServiceCollection services)
  {
      services.AddFeatureManagement()
              .AddFeatureFilter<MyCustomizedFilter>();
  }
}
```

> This tells the feature manager to use the "FeatureManagement" section from the configuration for feature flag settings. It also registers a feature filter named `MyCustomizedFilter`. When filters are referenced in feature flag settings (appsettings.json) the _Filter_ part of the type name can be omitted.

Reviewer: I'm thinking, perhaps we should omit the AddFeatureFilter part and leave that to the custom filters section.

Creating a feature filter provides a way to enable features based on criteria that you define. To implement a feature filter, the `IFeatureFilter` interface must be implemented. `IFeatureFilter` has a single method named `EvaluateAsync`. When a feature specifies that it can be enabled for a feature filter, the `EvaluateAsync` method is called. If `EvaluateAsync` returns `true` it means the feature should be enabled.

The following snippet demonstrates how to add a customized feature filter `MyCriteriaFilter`.

``` C#
services.AddFeatureManagement()
        .AddFeatureFilter<"MyCriteriaFilter">();
```

> Feature filters are registered through `AddFeatureFilter<T>` by the `IFeatureManagementBuilder` after `AddFeatureManagement` is called. These feature filters have access to the services that exist within the service collection that was used to add feature flags. Dependency injection can be used to retrieve these services.

Reviewer: ```suggestion
Feature filters are registered by calling `AddFeatureFilter<T>` on the `IFeatureManagementBuilder` returned from `AddFeatureManagement`. These feature filters have access to the services that exist within the service collection that was used to add feature flags. Dependency injection can be used to retrieve these services.

```

**Note:** Only a single feature filter interface can be implemented by a single type. Trying to add a feature filter that implements more than a single feature filter interface will result in an `ArgumentException`.

> **Advanced:** Filters pf `IFeatureFilter` and `IContextualFeatureFilter` can share the same alias. Specifically, you can have one filter alias shared by 0 or 1 IFeatureFilter and 0 or N IContextualFeatureFilter<T>, so long as there is at most 1 applicable filter for T.

Reviewer: Seems like it deserves it's own header.

Like ### Using contextual and non-contextual filters with the same name

The following passage describes the process of selecting a filter when contextual and non-contextual filters of the same name are registered in an application.

Let's say you have a non-contextual filter called `FilterA` and two contextual filters `FilterB` and FilterC which accept `TypeB` and `TypeC` contexts respectively. All of three filters share the same alias `SharedFilterName`.

You also have a feature flag `MyFeature` which uses the feature filter `SharedFilterName` in its configuration.

If all of three filters are registered:
* When you call IsEnabledAsync("MyFeature"), the `FilterA` will be used to evaluate the feature flag.
* When you call IsEnabledAsync("MyFeature", context), if context's type is `TypeB`, `FilterB` will be used and if context's type is `TypeC`, `FilterC` will be used.
* When you call IsEnabledAsync("MyFeature", context), if context's type is `TypeF`, `FilterA` will be used.

### Built-In Feature Filters

> There a few feature filters that come with the `Microsoft.FeatureManagement` package. There are four built-in feature filters: `PercentageFilter`, `TimeWindowFilter`, `ContextualTargetingFilter` and `TargetingFilter`. Except for `TargetingFilter`, the other three built-in feature filters will be added automatically once the package is registered.

Reviewer: ```suggestion
There a few feature filters that come with the `Microsoft.FeatureManagement` package. There are four built-in feature filters: `PercentageFilter`, `TimeWindowFilter`, `ContextualTargetingFilter` and `TargetingFilter`. All filters, except for the `TargetingFilter`, are added automatically when feature management is registered.

```
